warden/group/handler.go

```diff
@@ -384,27 +363,88 @@ func (h *Handler) AddGroupMembers(w http.ResponseWriter, r *http.Request, ps htt
 //       403: genericError
 //       500: genericError
 func (h *Handler) RemoveGroupMembers(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
-	var ctx = r.Context()
 	var id = ps.ByName("id")
 
+	if err := h.checkRequest(r, fmt.Sprintf(h.PrefixResource(GroupResource), id), "members.remove"); err != nil {
+		h.H.WriteError(w, r, err)
+		return
+	}
+
 	var m membersRequest
 	if err := json.NewDecoder(r.Body).Decode(&m); err != nil {
 		h.H.WriteError(w, r, errors.WithStack(err))
 		return
 	}
 
-	if _, err := h.W.TokenAllowed(ctx, h.W.TokenFromRequest(r), &firewall.TokenAccessRequest{
-		Resource: fmt.Sprintf(h.PrefixResource(GroupResource), id),
-		Action:   "members.remove",
-	}, Scope); err != nil {
+	if err := h.Manager.RemoveGroupMembers(id, m.Members); err != nil {
 		h.H.WriteError(w, r, err)
 		return
 	}
 
-	if err := h.Manager.RemoveGroupMembers(id, m.Members); err != nil {
+	w.WriteHeader(http.StatusNoContent)
+}
+
+// swagger:route PUT /warden/groups/{id}/members warden replaceMembersInGroup
+//
+// Replace the members of a group
+//
+// The subject making the request needs to be assigned to a policy containing:
+//
+//  ```
+//  {
+//    "resources": ["rn:hydra:warden:groups:<id>"],
+//    "actions": ["members.update"],
+//    "effect": "allow"
+//  }
+//  ```
+//
+//     Consumes:
+//     - application/json
+//
+//     Produces:
+//     - application/json
+//
+//     Schemes: http, https
+//
+//     Security:
+//       oauth2: hydra.warden.groups
+//
+//     Responses:
+//       204: emptyResponse
```

Comment:
`PUT` requests typically don't return 204 responses but instead 200 + body. 204 is typically used for `DELETE` responses to aknowledge that the resource is gone

Reply:
I may not have changed everything correctly when copying the comments

Reply:
It's also in the response, see `w.WriteHeader(http.StatusNoContent)`